codec/amino.go

```diff
@@ -17,29 +19,38 @@ import (
 var Cdc *Codec
 
 func init() {
-	cdc := New()
-	RegisterCrypto(cdc)
-	RegisterEvidences(cdc)
-	Cdc = cdc.Seal()
+	Cdc := New()
+	RegisterCrypto(Cdc)
+	RegisterEvidences(Cdc)
+	Cdc.Seal()
+}
+
+// Codec defines a wrapper for an Amino codec that properly handles protobuf
+// types with Any's
+type Codec struct {
```

Comment:
will this be deprecated in the future in favor of `AminoCodec`? are there any difference between the 2?

Reply:
In the ADR updates in this PR I noted that we would rename this to `LegacyAmino`. It is more or less deprecated. The difference between the two is that `Marshaler` expects types that implement `ProtoMarshaler` (i.e. generated by protoc). So any code using `Marshaler` is more or less ensured to be proto compatible. This legacy `Codec` is pure amino.